### PR TITLE
add basic CI checks

### PR DESCRIPTION
**Pull request by dtinth**

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Checks
+on:
```
Comment (beam41) [inline .github/workflows/checks.yml:2]: Shouldn't we check on push to main branch too?

Comment (Leomotors) [reply .github/workflows/checks.yml:2]: +1

File: src/events/interactionCreate.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export default {
       if (!bypass) return
       try {
         console.log(
-          `[Command] ${interaction.user.tag} (${interaction.user.id}) > ${interaction.commandName}`
+          `[Command] ${interaction.user.tag} (${interaction.user.id}) > ${interaction.commandName}`,
```
Comment (Leomotors) [inline src/events/interactionCreate.ts:15]: Should trailing comma be `es5`. Feel like too much comma here.

Comment (Leomotors) [reply src/events/interactionCreate.ts:15]: cc: @beam41

Comment (beam41) [reply src/events/interactionCreate.ts:15]: It fixes problem of someone change function signature for me, I admit it's too much sometimes 😅

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm install
+      - name: Build and lint
+        run: pnpm lint
```
Comment (beam41) [inline .github/workflows/checks.yml:19]: Should add another step for type check `tsc --noEmit -p .`

File: src/events/interactionCreate.ts (hunk 2 above)
Comment (beam41) [reply src/events/interactionCreate.ts:15]: Btw, prettier automatically add comma for me so I don't think about it much.

File: .github/workflows/checks.yml (hunk 1 above)
Comment (dtinth) [reply .github/workflows/checks.yml:2]: done

**Review by Gusb3ll — APPROVED** [review]
LGTM kubb

**Pull request merged**